compiler/src/main/scala/org/scalajs/nscplugin/GenJSExports.scala

```diff
@@ -697,57 +697,83 @@ trait GenJSExports[G <: Global with Singleton] extends SubComponent {
         previousArgsValues: Int => List[js.Tree])(
         implicit pos: Position): js.Tree = {
 
-      val trgSym = {
-        if (sym.isClassConstructor) {
+      val owner = sym.owner
+      val isNested = owner.isLifted && !isStaticModule(owner.originalOwner)
+
+      val (trgSym, trgTree) = {
+        if (!isNested && sym.isClassConstructor) {
+          assert(captures.isEmpty, "expected empty captures")
           /* Get the companion module class.
-           * For inner classes the sym.owner.companionModule can be broken,
-           * therefore companionModule is fetched at uncurryPhase.
+           * For classes nested inside of modules the sym.owner.companionModule
+           * can be broken, therefore companionModule is fetched at
+           * uncurryPhase.
+           */
+          val trgSym = enteringPhase(currentRun.namerPhase) {
+            owner.linkedClassOfClass
+          }
+          (trgSym, genLoadModule(trgSym))
+        } else if (!isNested && static) {
+          assert(captures.isEmpty, "expected empty captures")
+          (owner, genLoadModule(owner))
+        } else if (sym.isClassConstructor || static) {
+          assert(isNested, "unexpected isNested")
```

Comment:
I have re-structured the flow and added some explanations. Is it clearer now?

(I refrained from using a match, because the tests would get too far away from their values, so it'd be hard to see what case `(true, true, false)` is).

Reply:
Thank you; it's much clearer now.